### ISSUE TITLE
Contacts not syncing from Android phone

### ISSUE THREAD

**Issue by ericblank** [nextcloud/contacts#132]
### Steps to reproduce
1. Add URL to DavDroid
2. Attempt sync
3.

### Expected behaviour

Contacts sync to nextcloud
### Actual behaviour
Contacts do not sync, calendar works correctly. 

### Server configuration
**Operating system**:
Ubuntu
**Web server:**
Apache
**Database:**
MySQL
**PHP version:**

**Nextcloud version:** (see Nextcloud admin page)
11
**Contacts version:** (see Nextcloud apps page)

**Updated from an older Nextcloud or fresh install:**
New
**Signing status:**

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```

**The content of config/config.php:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**

**Operating system:**

**CardDAV-clients:**

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by skjnldsv** [comment]
Please update your issue with the correct requested informations. We can't help you with so little data.

**Comment by DIY89** [comment]
**Server configuration**

**Operating system:**  `Debian-Jessie`
**Web server:** `Apache`
**Database:** `MySQL`
PHP version: `5.6.29`
**Nextcloud version:** `11.0.1 (stable)`
**Contacts version:** `1.5.3`

**Signing status:** `No errors have been found.`

**List of activated apps:**
```
Enabled:
  - activity: 2.4.1
  - bookmarks: 0.9.1
  - calendar: 1.5.0
  - comments: 1.1.0
  - contacts: 1.5.3
  - dav: 1.1.1
  - federatedfilesharing: 1.1.1
  - federation: 1.1.1
  - files: 1.6.1
  - files_pdfviewer: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - firstrunwizard: 2.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - survey_client: 0.1.5
  - systemtags: 1.1.3
  - tasks: 0.9.4
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - admin_audit
  - encryption
  - external
  - files_accesscontrol
  - files_automatedtagging
  - files_external
  - files_retention
  - templateeditor
 -  user_external
  - user_ldap
  - user_saml

```

**The content of config/config.php:**
```
{
    "system": {
        "instanceid": "oc9mx64qkbcl",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "trusted_domains": [
            "smarthomediy.ddns.net"
        ],
        "datadirectory": "\/home\/pi\/usb\/nextcloud",
        "overwriteprotocol": "https",
        "overwrite.cli.url": "https:\/\/smarthomediy.ddns.net",
        "dbtype": "mysql",
        "version": "11.0.1.2",
        "dbname": "cloudDB",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "maintenance": false
    }
}
```

**External Storage (yes):** `/home/pi/usb/nextcloud`

**Encryption:** `yes`

**External User-Backend**: `No`

**Client configuration:** `Browser: "Firefox 51.01"`
**Operating system:** `Win 10 64bit`

**CardDAV-clients:** `CardDAV-Sync free (Android 4.1.1)`

**Logs:
Web server error log:**
```

{"reqId":"3Cp6wjuo6AEuHd3Ym0tI","remoteAddr":"ip-address","app":"no app in context","message":"Exception: {\"Exception\":\"InvalidArgumentException\",\"Message\":\"The given object name is invalid\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/activity\\\/lib\\\/FilesHooks.php(892): OC\\\\Activity\\\\Event->setObject('files', 2, false)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/activity\\\/lib\\\/FilesHooks.php(184): OCA\\\\Activity\\\\FilesHooks->addNotificationsForUser('adminuser', 'created_self', Array, 2, false, true, true, 0, 'file_created')\\n#2 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/activity\\\/lib\\\/FilesHooks.php(111): OCA\\\\Activity\\\\FilesHooks->addNotificationsForFileAction('\\\/', 'file_created', 'created_self', 'created_by')\\n#3 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/activity\\\/lib\\\/FilesHooksStatic.php(45): OCA\\\\Activity\\\\FilesHooks->fileCreate('\\\/')\\n#4 [internal function]: OCA\\\\Activity\\\\FilesHooksStatic::fileCreate(Array)\\n#5 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/hook.php(106): call_user_func(Array, Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Files\\\/View.php(1237): OC_Hook::emit('OC_Filesystem', 'post_create', Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Files\\\/View.php(1163): OC\\\\Files\\\\View->runHooks(Array, '\\\/adminuser\\\/fil...', true)\\n#8 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Files\\\/View.php(266): OC\\\\Files\\\\View->basicOperation('mkdir', '\\\/adminuser\\\/fil...', Array)\\n#9 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Files\\\/Node\\\/Folder.php(149): OC\\\\Files\\\\View->mkdir('\\\/adminuser\\\/fil...')\\n#10 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Files\\\/Node\\\/Root.php(380): OC\\\\Files\\\\Node\\\\Folder->newFolder('\\\/adminuser\\\/fil...')\\n#11 [internal function]: OC\\\\Files\\\\Node\\\\Root->getUserFolder('adminuser')\\n#12 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Files\\\/Node\\\/LazyRoot.php(65): call_user_func_array(Array, Array)\\n#13 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Files\\\/Node\\\/LazyRoot.php(282): OC\\\\Files\\\\Node\\\\LazyRoot->__call('getUserFolder', Array)\\n#14 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Server.php(947): OC\\\\Files\\\\Node\\\\LazyRoot->getUserFolder('adminuser')\\n#15 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Session.php(411): OC\\\\Server->getUserFolder('adminuser')\\n#16 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Session.php(479): OC\\\\User\\\\Session->prepareUserLogin(true)\\n#17 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Session.php(299): OC\\\\User\\\\Session->loginWithPassword(*** sensitive parameters replaced ***)\\n#18 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Setup.php(393): OC\\\\User\\\\Session->login(*** sensitive parameters replaced ***)\\n#19 \\\/var\\\/www\\\/nextcloud\\\/core\\\/Controller\\\/SetupController.php(65): OC\\\\Setup->install(Array)\\n#20 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/base.php(957): OC\\\\Core\\\\Controller\\\\SetupController->run(Array)\\n#21 \\\/var\\\/www\\\/nextcloud\\\/index.php(40): OC::handleRequest()\\n#22 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Activity\\\/Event.php\",\"Line\":401}","level":3,"time":"2017-01-25T15:08:34+00:00","method":"POST","url":"\/index.php","user":"adminuser","version":"11.0.1.2"}
{"reqId":"loFzUmKRMDCSFURnQkop","remoteAddr":"ip-address","app":"core","message":"Login failed: 'user' (Remote IP: 'ip-address')","level":2,"time":"2017-01-25T15:08:54+00:00","method":"PROPFIND","url":"\/remote.php\/webdav\/","user":"--","version":"11.0.1.2"}
```

**Browser log:**

```
Fatal	webdav	Sabre\DAV\Exception\ServiceUnavailable: HTTP/1.1 503 System in maintenance mode.

    [internal function] OCA\DAV\Connector\Sabre\MaintenancePlugin->checkMaintenanceMode(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
    /var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php - line 105: call_user_func_array(Array, Array)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 466: Sabre\Event\EventEmitter->emit('beforeMethod', Array)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 254: Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
    /var/www/nextcloud/apps/dav/appinfo/v1/webdav.php - line 60: Sabre\DAV\Server->exec()
    /var/www/nextcloud/remote.php - line 165: require_once('/var/www/nextcl...')
```

```   
Fatal	webdav	Sabre\DAV\Exception\MethodNotAllowed: HTTP/1.1 405 The resource you tried to create already exists

    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php - line 590: Sabre\DAV\Server->createCollection('addressbooks/us...', Object(Sabre\DAV\MkCol))
    [internal function] Sabre\DAV\CorePlugin->httpMkcol(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
    /var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php - line 105: call_user_func_array(Array, Array)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 479: Sabre\Event\EventEmitter->emit('method MKCOL', Array)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 254: Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
    /var/www/nextcloud/apps/dav/lib/Server.php - line 227: Sabre\DAV\Server->exec()
    /var/www/nextcloud/apps/dav/appinfo/v2/remote.php - line 30: OCA\DAV\Server->exec()
    /var/www/nextcloud/remote.php - line 165: require_once('/var/www/nextcl...')

```

**Firefox-Debugger:**
```
JQMIGRATE: Migrate is installed, version 1.4.0  jquery-migrate.min.js:2:542
getAttributeNode() sollte nicht mehr verwendet werden. Verwenden Sie stattdessen getAttribute().  jquery-migrate.min.js:2:1714
Das nodeValue-Attribut von Attributen sollte nicht mehr verwendet werden. Verwenden Sie value stattdessen.  jquery-migrate.min.js:2:1714
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src 'nonce-Y0FRcExtbzAwMEdTZlQ5UnZZbUIwN2dvQ01pMWtTQ0hkT1JjSlRLM0tlUT06Q2tGZ2JSTUU0eEdtQjFRSng3cjBsOUpkUElIWCtYbnBKTkZzVXdXQ0c0Yz0=' 'unsafe-eval'").  contacts

```

**Comment by voxdemonix** [comment]
i have same problem with DavDroid too (on android) :
- i can't export contact from my android
- i can't export Group from my android 
- i can't modify any nextcloud contact via my Android (i can modify only via WEBUI)
- i can just import contact
- creat a group on nextcloud with same name than a group on my android = creat duplicate group on my android (no fusion)
- i can't adjust contact's pics

In davdroid i haved already creat an account for calendar and an account for contact.

**Comment by blubbsy** [comment]
Same problem for me. no integrity problems, normal syncing works well, but i cant sync with davdroid (and no error is showing up...)

**Comment by gusarg81** [comment]
Hi,

Same for me, just buyed Davdroid and does not sync contacs from android to netxcloud and either otherwise.

Desinstalled Davdoird (and ask for refund) till this can be fixed. Thanks!

**Comment by dhowe** [comment]
same here

**Comment by gusarg81** [comment]
Hi, I've attached logs in my comment: https://github.com/nextcloud/contacts/issues/132#issuecomment-310911198

**Comment by skjnldsv** [comment]
Is this issue still occuring? I'musing DavDroid myself and everything is syncing fine.
@DIY89 your installation seems to be in maintenance mode, so everything is disabled, please see your nextcloud config file.

@gusarg81 Could you check your password and login? Or try to create an app password specificely for davdroid (see https://example.org/index.php/settings/personal#apppasswords

**Comment by gusarg81** [comment]
I've forgot to add that I've figured it out and fixed this problem this past weekend, which happens on Xiaomi phones: https://davdroid.bitfire.at/faq/entry/miui-no-synchronization/

Now calendars and contacts are synced!

Thanks.

**Comment by skjnldsv** [comment]
Thank you!

**Comment by dhowe** [comment]
I have the same issue on a OnePlus 3T...

**Comment by gusarg81** [comment]
@dhowe check the link of Davdroid posted, maybe It can help you as well.

**Comment by skjnldsv** [comment]
@dhowe, could you open your own issue with your own logs? :)

**Comment by dhowe** [comment]
sure, though actually my problem seems a bit different, as my contacts don't sync from any device, neither laptop nor phone (calendar works as expected)

**Comment by dhowe** [comment]
The link is not relevant to the OnePlus -- if I can narrow down the issue a bit, I will create a new issue

**Comment by forreggbor** [comment]
I use OpenSync Free. Connection to the server works. If I create new Carddav or Caldav entry in Opensync those are synced to and appears on the server. But there are no content at all. If I import my contact from vcard file it appears in nextcloud but after I do a sync from phone all contact disappears from nextcloud.

**Comment by jirislav** [comment]
I use CardDAV free & I've had the same problem, which @gusarg81 [posted](https://davdroid.bitfire.at/faq/entry/miui-no-synchronization/) - the problem was in MIUI at my XIAOMI phone, solved with adding a permission to Autostart, see

**Comment by D4rkPh0en1x** [comment]
I have a similar problem just with one difference:
On my Huawei P9 and iPAD Pro all works fine.
Nowe my wife got a new Huawei P10 and I configured like the other Android device with DavDroid but nothing syncs. I checked the DavDroid logs and there he tells me access denied to contact and calendar. In NextCloud logs I see nothing. 
I also can access the files with the Nextcloud App, that works fine. It’s just the sync with DavDroid not working,,,
